tests/semantic_analyser.cpp

```diff
@@ -4371,6 +4410,20 @@ BEGIN { @map[0] = 1; for ($kv : @map) { print($kv); } }
              false);
 }
 
+TEST(semantic_analyser, for_loop_castable_map_missing_feature)
+{
+  test_error("BEGIN { @map[0] = count(); for ($kv : @map) { print($kv); } }",
```

Comment:
I thought it might be overkill to test all castable map types here (I check them all above in 'castable_map_missing_feature') but I can add if people want. I can also try something dynamic.